internal/pkg/cli/pipeline_init.go

```diff
@@ -247,14 +247,36 @@ func (o *initPipelineOpts) createPipelineProvider() (manifest.Provider, error) {
 	return manifest.NewProvider(config)
 }
 
+func (o *initPipelineOpts) getEnvFromCache(environmentName string) (*config.Environment, error) {
+	var environment *config.Environment
+	for _, env := range o.envs {
+		if env.Name == environmentName {
+			environment = env
+		}
+	}
+	if environment == nil {
+		return nil, fmt.Errorf("environment %s in application %s is not found", environmentName, o.AppName())
+	}
+	return environment, nil
+}
+
 func (o *initPipelineOpts) createPipelineManifest() error {
 	pipelineName := o.createPipelineName()
 	provider, err := o.createPipelineProvider()
 	if err != nil {
 		return fmt.Errorf("create pipeline provider: %w", err)
 	}
 
-	manifest, err := manifest.CreatePipeline(pipelineName, provider, o.Environments)
+	var environments []*config.Environment
+	for _, environmentName := range o.Environments {
+		env, err := o.getEnvFromCache(environmentName)
+		if err != nil {
+			return err
+		}
+		environments = append(environments, env)
+	}
+
+	manifest, err := manifest.CreatePipeline(pipelineName, provider, environments)
```

Comment:
Instead of passing `[]*config.Environment` as input to this function, what do you think of passing `[]manifest.PipelineStage`? so this method would transform `[]*config.Environment -> []manifest.PipelineStage`.

Since the function is in the `manifest` pkg, accepting `[]manifest.PipelineStage` will make it more flexible. The `PipelineStage` can be created from a `config.Environment` object or some other way.

Reply:
I addressed. However, now TestNewPipelineManifest just become meaningless because this test expects PipelineStage. I plan to change this test such as it expects PipelineManifest. If you have any opinion, please give me a feedback!

Reply:
hmm yeah I agree, the only behavior that might be interesting to test in `TestNewPipelineManifest` is checking if an error is returned if the list is empty and that's it

Reply:
Oh, does it mean we simply need to remove happy case and retain only error case? I just modified happy case with `PipelineManifest` as expected by imitating backend_svc_test.go. If I need to remove happy case, please point out!